pkg/api/types.go

```diff
@@ -195,6 +195,119 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
```

Comment:
I don't understand why we need both this and VolumeSource.  Can't we just re-use PersistentVolumeSource?
Along the same lines, I think that the distinction between a user and an administrator here is somewhat artificial and temporary, and should not be unnecessarily exposed via the API.

In future I imagine that both users and administrators will be able to create PD's, and should both use the same API.